builtin/game/chat.lua

```diff
@@ -418,121 +418,111 @@ core.register_chatcommand("remove_player", {
 	end,
 })
 
+
+-- pos may be a non-integer position
+local function find_free_position_near(pos)
+	local tries = {
+		{x=1, y=0, z=0},
+		{x=-1, y=0, z=0},
+		{x=0, y=0, z=1},
+		{x=0, y=0, z=-1},
+	}
+	for _, d in ipairs(tries) do
+		local p = vector.add(pos, d)
+		local n = core.get_node_or_nil(p)
+		if n then
+			local def = core.registered_nodes[n.name]
+			if def and not def.walkable then
+				return p
+			end
+		end
+	end
+	return pos
+end
+
+-- Teleports player <name> to <p> if possible
+local function teleport_to_pos(name, p)
+	local lm = 31000
+	if p.x < -lm or p.x > lm or p.y < -lm or p.y > lm
+			or p.z < -lm or p.z > lm then
+		return false, "Cannot teleport out of map bounds!"
+	end
+	local teleportee = core.get_player_by_name(name)
+	if not teleportee then
+		return false, "Cannot get player with name " .. name
+	end
+	if teleportee:get_attach() then
+		return false, "Cannot teleport, " .. name ..
+			" is attached to an object!"
+	end
+	teleportee:set_pos(p)
+	return true, "Teleporting " .. name .. " to " .. core.pos_to_string(p)
+end
+
+-- Teleports player <name> next to player <target_name> if possible
+local function teleport_to_player(name, target_name)
+	if name == target_name then
+		return false, "One does not teleport to oneself."
+	end
+	local teleportee = core.get_player_by_name(name)
+	if not teleportee then
+		return false, "Cannot get teleportee with name " .. name
+	end
+	if teleportee:get_attach() then
+		return false, "Cannot teleport, " .. name ..
+			" is attached to an object!"
+	end
+	local target = core.get_player_by_name(target_name)
+	if not target then
+		return false, "Cannot get target player with name " .. target_name
+	end
+	local p = find_free_position_near(target:get_pos())
+	teleportee:set_pos(p)
+	return true, "Teleporting " .. name .. " to " .. target_name .. " at " ..
+		core.pos_to_string(p)
```

Comment:
You should round first

```suggestion
		core.pos_to_string(vector.round(p))
```

Reply:
`core.pos_to_string(p, 1)`

Reply:
Thanks